packages/did/src/DidDetails/FullDidDetails.utils.ts

```diff
@@ -28,10 +28,32 @@ const methodMapping: Record<string, VerificationKeyRelationship | undefined> = {
   web3Names: 'authentication',
 }
 
-export function getKeyRelationshipForExtrinsic(
-  extrinsic: Extrinsic
+export function getKeyRelationshipForExtrinsic<A extends AnyTuple = AnyTuple>(
+  call: CallBase<A>
 ): VerificationKeyRelationship | undefined {
-  const { section, method } = extrinsic.method
+  const { section, method } = call
+
+  // get the VerificationKeyRelationship of a batched call
+  if (
+    section === 'utility' &&
+    (method === 'batch' || method === 'batchAll' || method === 'forceBatch') &&
+    call.args[0].toRawType() === 'Vec<Call>'
+  ) {
+    // map all calls to their VerificationKeyRelationship and deduplicate the items
+    const keys = new Set(
+      (call.args[0] as any as Array<CallBase<A>>).map(
```

Comment:
That might be a bit controversial? I guess the correct way would be to use ´registry.createType`? AFAIK for that we would need this function to be `async`.